modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/IgniteSqlValidator.java

```diff
@@ -1559,11 +1507,11 @@ private void setDynamicParamType(SqlDynamicParam dynamicParam, RelDataType dataT
      * this method throws {@link IllegalArgumentException}.
      */
     @Nullable
-    private Object getDynamicParamValue(SqlDynamicParam dynamicParam) {
+    private RelDataType getDynamicParamType(SqlDynamicParam dynamicParam) {
         int index = dynamicParam.getIndex();
         DynamicParamState paramState = dynamicParameters.computeIfAbsent(index, (i) -> new DynamicParamState());
```

Comment:
Can we create a constant `DynamicParamState.NULL = new DynamicParamState()` and replace `computeIfAbsent` with `getOrDefault(index, NULL)` ?